tests/tstackarrays.nim

```diff
@@ -34,3 +34,27 @@ suite "Stack arrays":
     expect RangeError:
       arr[3] = "another test"
 
+  test "proof of stack allocation":
+    proc fun() =
+      # NOTE: has to be inside a proc otherwise x1 not allocated on stack.
+      var x1 = 0
+      var arr = allocStackArray(int, 3)
+      var x2 = 0
+      let p_addr = cast[int](addr arr)
+      let p_x1 = cast[int](addr x1)
+      let p_x2 = cast[int](addr x2)
+      check:
+        # stack can go either up or down
+        p_addr > min(p_x1, p_x2) and p_addr < max(p_x1, p_x2)
+    fun()
+
+  test "skip initialization":
```

Comment:
I don't think this test is needed at all. Firstly, it doesn't (and can't reliably) check for arrays being not initialized. Secondly, what's the point in comparing two arrays from subsequent calls? We don't guarantee that they're going to be the same.